test/e2e_node/services/kubelet.go

```diff
@@ -260,6 +320,52 @@ func (e *E2EServices) startKubelet() (*server, error) {
 	return server, server.start()
 }
 
+// writeKubeletConfigFile writes the kubelet config file based on the args and returns the filename
+func writeKubeletConfigFile(internal *kubeletconfig.KubeletConfiguration, path string) error {
+	// extract the KubeletConfiguration and convert to versioned
+	versioned := &v1alpha1.KubeletConfiguration{}
+	scheme, _, err := scheme.NewSchemeAndCodecs()
+	if err != nil {
+		return err
+	}
+	if err := scheme.Convert(internal, versioned, nil); err != nil {
+		return err
+	}
+	// encode
+	encoder, err := newKubeletConfigJSONEncoder()
+	if err != nil {
+		return err
+	}
+	data, err := runtime.Encode(encoder, versioned)
+	if err != nil {
+		return err
+	}
+	// create the directory, if it does not exist
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return err
+	}
+	// write the file
+	if err := ioutil.WriteFile(path, data, 0755); err != nil {
+		return err
+	}
+	return nil
+}
+
+func newKubeletConfigJSONEncoder() (runtime.Encoder, error) {
```

Comment:
There is a helper in util.go, right?

Reply:
ah, yes

Reply:
oh, import cycle to use anything in util from services, apparently:
```
# k8s.io/kubernetes/test/e2e_node
import cycle not allowed in test
package k8s.io/kubernetes/test/e2e_node (test)
	imports k8s.io/kubernetes/test/e2e_node/services
	imports k8s.io/kubernetes/test/e2e_node
```

Reply:
OK. I'm fine with this for now. We could probably add a new package, `kubeconfig_util` or something just put kubelet config util functions there, so that both test and services can use it.
But I'm fine with doing that in the future when we have more duplicated kube config related util functions.